docs/tasks/administer-cluster/running-cloud-controller.md

```diff
@@ -1,38 +1,96 @@
+
+
+## Examples
+
+If you are using a cloud that is currently supported in Kubernetes core and would like to adopt cloud controller manager, see the [cloud controller manager in kubernetes core](https://github.com/kubernetes/kubernetes/tree/master/cmd/cloud-controller-manager).
+
+For cloud controller managers not in Kubernetes core, you can find the respective projects in repos maintained by cloud vendors or sig leads.
+
+* [DigitalOcean](https://github.com/digitalocean/digitalocean-cloud-controller-manager)
+* [Rancher](https://github.com/rancher/rancher-cloud-controller-manager)
+
+For providers already in Kubernetes core, you can run the in-tree cloud controller manager as a [Deployment](/docs/tasks/administer-cluster/cloud-controller-manager-deployment-example.yaml) or [Daemonset](/docs/tasks/administer-cluster/cloud-controller-manager-daemonset-example.yaml) in your cluster.
+
+
+## Limitations
+
+Running cloud controller manager comes with a possible limitations. Although these limitations are being in addressed in upcoming releases, it's important that you are aware of these limitations for production workloads.
+
+### Support for Volumes
+
+Cloud controller manager does not implement any of the volume controllers found in `kube-controller-manager` as volume integrations also require coordination with kubelets. As we evolve CSI (container storage interface) and add stronger support for flex volume plugins, necessary support will be added to cloud controller manager so that clouds can fully integrate with volumes.
+
+### Scalability
+
+In the previous architecture for cloud providers, we relied on kubelets using a local metadata service to retreive node information about itself. With this new architecture, we now fully rely on the cloud controller managers to retrieve information for all nodes. For very larger clusters, you should consider possible bottle necks such as resource requirements and API rate limiting.
+
+### Chicken and Egg
+
+The goal of the cloud controller manager project is to decouple development of cloud features from the core Kubernetes project. Unforunately, many aspects of the Kubernetes project has assumptions that cloud provider features are tightly integrated into the project. As a result, adopting this new architecture can create several situations where a requests is being made for information from a cloud provider, but the cloud controller manager may not be able to return that information without the original request being complete.
+
+A good example of this is the TLS bootstrapping feature in the Kubelet. Currently, TLS bootstraping assumes that the Kubelet has the ability to ask the cloud provider (or a local metadata service) for all its address types (private, public, etc) but cloud controller manager cannot set a node's address types without being initialzed in the first place which requires that the kubelet has TLS certificates to communicate with the apiserver.
```

Comment:
Is there a ticket tracking the TLS bootstrap issue? All i can find in the [beta ticket](https://github.com/kubernetes/kubernetes/issues/48690) is a link to a comment. It'd be nice to link here to provide more info if the user wants it as well as a place to subscribe for updates.

Reply:
cc @liggitt

Reply:
Don't have that yet I think.
AI for the WG I guess?